docs/uncertainty/index.rst

```diff
@@ -0,0 +1,353 @@
+.. |distribution| replace:: :class:`~astropy.uncertainty.Distribution`
+
+.. _astropy-uncertainty:
+
+*******************************************************
+Uncertainties and Distributions (`astropy.uncertainty`)
+*******************************************************
+
+.. warning::
+
+    `astropy.uncertainty` is currently a work-in-progress, and thus it is quite
+    possible there will be API changes in later versions of Astropy. If you have
+    specific ideas for how it might be improved, please  let us know on the
+    `astropy-dev mailing list`_ or at http://feedback.astropy.org .
+
+Introduction
+============
+
+In addition to |quantity|, astropy provides a |distribution| object to represent
+statistical distributions in a form that acts as a drop-in replacement for
+|quantity| or regulay Numpy arrays. Used in this manner, |distribution| provides
+uncertainty propogation at the cost of additional computation.  It can also
+more generally represent sampled distributions for e.g., Monte Carlo calculation
+techniques.
+
+The core object for this feature is the |distribution|.  Currently, all
+such distributions are Monte Carlo sampled.  While this means each distribution
+may take more memory, it allows arbitrarily complex operations to be performed
+on distributions while maintaining their correlation structure. Some specific
+well-behaved distributions (e.g., the Normal distribution) have
+analytic forms which may eventually enable a more compact/efficient
+representation, but this is not currently implemented.
+
+Getting Started
+===============
+
+To demonstrate a simple use case for distributions, consider the problem of
+uncertainty propogation of normal distributions.  Assume there are two
+measurements you wish to add.  We start with some initial imports/setup::
+
+  >>> import numpy as np
+  >>> from astropy import units as u
+  >>> from astropy import uncertainty as unc
+  >>> np.random.seed(12345)  # ensures reproducible example numbers
+
+Now we create two |distribution| objects to represent our distributions::
+
+  >>> a = unc.normal(1*u.kpc, std=30*u.pc)
+  >>> b = unc.normal(2*u.kpc, std=40*u.pc)
+
+For normal distributions, the centers should add as expected, and the standard
+deviations add in quadrature.  We can check these results (to the limits of our
+Monte Carlo sampling) trivially with |distribution| arithmetic and attributes::
+
+  >>> c = a + b
+  >>> c # doctest: +ELLIPSIS
+  <QuantityDistribution [..., 3.06148029,...] kpc with n_samples=1000>
+  >>> c.pdf_mean # doctest: +FLOAT_CMP
+  <Quantity 3.0005627 kpc>
+  >>> c.pdf_std.to(u.pc) # doctest: +FLOAT_CMP
+  <Quantity 51.4783738 pc>
+
+Indeed these are close to the expectations. While this may seem unnecessary for
+the simple Gaussian case, for more complex distributions or airthmetic
+operations where error analysis becomes untenable, |distribution| still powers
+through:
+
+  >>> d = unc.poisson(3*u.kpc)

```

Comment:
Noooooooooooooooooooooooo

Reply:
In my defense, I did not just do this blindly: I was considering a case where you actually have binned (i.e., discrete) data binned in kpc.  But I see your point that that is *not* a good intro example.